pkg/pillar/cmd/client/client.go

```diff
@@ -516,42 +575,98 @@ func selfRegister(zedcloudCtx zedcloud.ZedCloudContext, tlsConfig *tls.Config, d
 		log.Errorf("%s\n", string(contents))
 		done = false
 	}
-	return done
+
+	return done, rtf == types.ExtStatusCertUnknownAuthority
+}
+
+// fetch V2 certs from cloud, return GotCloudCerts and ServerIsV1 boolean
+// if got certs, the leaf is saved to types.ServerCertName file
```

Comment:
I don't see any code in this function to save to that file. Odd.

Reply:
part of the verify in this version. could move out to the caller as you mentioned.